app/services/request_generator.rb

```diff
@@ -0,0 +1,46 @@
+# typed: true
+# frozen_string_literal: true
+
+# This generates a RequestDRO for a work
+class RequestGenerator
+  extend T::Sig
+
+  sig { params(work: Work).returns(Cocina::Models::RequestDRO) }
+  def self.generate_model(work:)
+    new(work: work).generate_model
+  end
+
+  sig { params(work: Work).void }
+  def initialize(work:)
+    @work = work
+  end
+
+  def generate_model
+    Cocina::Models::RequestDRO.new(generate)
+  end
+
+  sig { returns(Hash) }
+  # rubocop:disable Metrics/MethodLength
+  def generate
+    {
+      administrative: {
+        hasAdminPolicy: 'druid:pq757cd0790' # TODO: What should this be? this is the hydrus APO.
+      },
+      identification: {
+        sourceId: "hydrus:#{work.id}" # TODO: what should this be?
```

Comment:
@andrewjbtw can you give feedback on this?

Reply:
For the sourceId we should generate it the way we do for Hydrus, though I guess we shouldn't use the word "Hydrus" anymore. But the auto-generated source we've been using has been working fine.

For APO it's more complicated and we should discuss. There are drawbacks to the current method but changing it could cause unexpected problems.